### PR TITLE
The 'delete' button should be inactive when the circle is active

### PR DESCRIPTION
**Pull request by luanecavalcantizup**
## Problem

The `delete` button should be inactive/disabled when the circle is active or when it is a default circle (active or inactive)

## Solution

-[frontend] Disable `delete` button when circle is active or when it is a default circle (be it active or inactive) and show a tooltip giving information to user.
-[frontend] Disable `delete` button when a circle is deploying or undeploying.
-[backend] Add rule to backend to block deleting a default circle

## Before & After Screenshots

**BEFORE**:
Active default circle or other active circle can be deleted
image

**AFTER**:
Active default circle or other active circle cannot be deleted and shows a tooltip
Captura de Tela (339)

### CODE REVIEW

File: ui/src/modules/Circles/Comparation/Item/helpers.ts [hunk 1]
```diff
@@ -89,14 +89,18 @@ export const getActiveMetricDescription = (activeMetricType: METRICS_TYPE) => {
...
   const cannotDeleteDefaultCircleMessage =
     'Default circle is deployed to all<br /> users, so it cannot be deleted.';
+  const cannotDeleteInactiveDefaultCircleMessage =
+    'Default circle cannot be deleted.';
   let tooltipMessage = '';
 
-  if (isDefaultCircle(circleName)) {
+  if (isDefaultCircle(circle?.name) && !hasDeploy(circle)) {
+    tooltipMessage = cannotDeleteInactiveDefaultCircleMessage;
+  } else if (isDefaultCircle(circle?.name) && hasDeploy(circle)) {
```
Comment (pedroborgeszup) [inline ui/src/modules/Circles/Comparation/Item/helpers.ts:103]: These 2 conditions above can be reduced to just 1. Ex. If (isDefaultCircle (circle? .Name))

Comment (luanecavalcantizup) [reply ui/src/modules/Circles/Comparation/Item/helpers.ts:103]: I think having different messages for when default circle is active and when is inactive is more descriptive. Jose (the UX) did the design in invision for these cases. What do you think?

Comment (pedroborgeszup) [reply ui/src/modules/Circles/Comparation/Item/helpers.ts:103]: Ok! 👍

File: moove/web/src/main/resources/locale/messages.properties [hunk 2]
```diff
@@ -19,6 +19,8 @@ invalid.authorization=Failed on token authorization.
 move.build.invalid.column=Invalid column to move build.
 git.credentials.not.found=Git credentials not found.
 delete.deployed.build=Build can not be deleted, because it is already deployed in a circle.
+deploymet.active= There is an active deployment for the circle you want to delete
```
Comment (monicaribeiro) [inline moove/web/src/main/resources/locale/messages.properties:22]: This circle has an active deployment.

File: moove/web/src/main/resources/locale/messages_pt_BR.properties [hunk 3]
```diff
@@ -18,6 +18,8 @@ invalid.authorization=Falha no token de autorização.
 move.build.invalid.column=Falha ao mover o build para a coluna (coluna inválida).
 git.credentials.not.found=Credenciais git não encontradas.
 delete.deployed.build=O build não pode ser deletado pois está presente em um círculo.
+deployment.active=Existe um deployment ativo para o círculo que deseja apagar
```
Comment (monicaribeiro) [inline moove/web/src/main/resources/locale/messages_pt_BR.properties:21]:
```suggestion
deployment.active=Este círculo possui uma implantação ativa.
```

**Review by monicaribeiro — CHANGES_REQUESTED** [review]

**Review by monicaribeiro — DISMISSED** [review]

**Review by Leandroqo — DISMISSED** [review]

**Review by monicaribeiro — APPROVED** [review]

**Review by Leandroqo — APPROVED** [review]

**Review by monicaribeiro — DISMISSED** [review]

**Review by Leandroqo — APPROVED** [review]

**Review by monicaribeiro — APPROVED** [review]

**Pull request merged**